src/sub.c

```diff
@@ -89,21 +89,23 @@ _initOwnDispatcher(natsSubscription *sub)
     return NATS_UPDATE_ERR_STACK(s);
 }
 
-static inline void _cleanupOwnDispatcher(natsSubscription *sub)
+static inline void
+_cleanupOwnDispatcher(natsSubscription *sub)
 {
     nats_destroyQueuedMessages(&sub->ownDispatcher.queue);
 
     if (sub->ownDispatcher.thread != NULL)
     {
-        natsThread_Join(sub->ownDispatcher.thread);
+        natsThread_Detach(sub->ownDispatcher.thread);
```

Comment:
@kozlovic this was an oversight on my end I think since this code can be executed in the dispatch thread itself. Anyway, this restores the original logic. It did not appear to affect the test failure here, the wait in the test did.